sdks/universal-router-sdk/src/utils/routerCommands.ts

```diff
@@ -42,35 +53,99 @@ const PERMIT_BATCH_STRUCT =
-  [CommandType.PERMIT2_PERMIT_BATCH]: [PERMIT_BATCH_STRUCT, 'bytes'],
-  [CommandType.PERMIT2_TRANSFER_FROM]: ['address', 'address', 'uint160'],
-  [CommandType.PERMIT2_TRANSFER_FROM_BATCH]: [PERMIT2_TRANSFER_FROM_BATCH_STRUCT],
+  [CommandType.PERMIT2_PERMIT]: [
+    { name: 'permit', type: PERMIT_STRUCT },
+    { name: 'signature', type: 'bytes' },
+  ],
+  [CommandType.PERMIT2_PERMIT_BATCH]: [
+    { name: 'permit', type: PERMIT_BATCH_STRUCT },
+    { name: 'signature', type: 'bytes' },
+  ],
+  [CommandType.PERMIT2_TRANSFER_FROM]: [
+    { name: 'token', type: 'address' },
+    { name: 'recipient', type: 'address' },
+    { name: 'amount', type: 'uint160' },
+  ],
+  [CommandType.PERMIT2_TRANSFER_FROM_BATCH]: [
+    {
+      name: 'transferFrom',
+      type: PERMIT2_TRANSFER_FROM_BATCH_STRUCT,
+    },
+  ],
 
   // Uniswap Actions
-  [CommandType.V3_SWAP_EXACT_IN]: ['address', 'uint256', 'uint256', 'bytes', 'bool'],
-  [CommandType.V3_SWAP_EXACT_OUT]: ['address', 'uint256', 'uint256', 'bytes', 'bool'],
-  [CommandType.V2_SWAP_EXACT_IN]: ['address', 'uint256', 'uint256', 'address[]', 'bool'],
-  [CommandType.V2_SWAP_EXACT_OUT]: ['address', 'uint256', 'uint256', 'address[]', 'bool'],
-  [CommandType.V4_SWAP]: ['bytes'],
+  [CommandType.V3_SWAP_EXACT_IN]: [
+    { name: 'recipient', type: 'address' },
+    { name: 'amountIn', type: 'uint256' },
+    { name: 'amountOutMin', type: 'uint256' },
+    { name: 'path', subparser: Subparser.V3PathExactIn, type: 'bytes' },
+    { name: 'payerIsUser', type: 'bool' },
+  ],
+  [CommandType.V3_SWAP_EXACT_OUT]: [
+    { name: 'recipient', type: 'address' },
+    { name: 'amountOut', type: 'uint256' },
+    { name: 'amountInMax', type: 'uint256' },
+    { name: 'path', subparser: Subparser.V3PathExactOut, type: 'bytes' },
+    { name: 'payerIsUser', type: 'bool' },
+  ],
+  [CommandType.V2_SWAP_EXACT_IN]: [
+    { name: 'recipient', type: 'address' },
+    { name: 'amountIn', type: 'uint256' },
+    { name: 'amountOutMin', type: 'uint256' },
+    { name: 'path', type: 'address[]' },
+    { name: 'payerIsUser', type: 'bool' },
+  ],
+  [CommandType.V2_SWAP_EXACT_OUT]: [
+    { name: 'recipient', type: 'address' },
+    { name: 'amountOut', type: 'uint256' },
+    { name: 'amountInMax', type: 'uint256' },
+    { name: 'path', type: 'address[]' },
+    { name: 'payerIsUser', type: 'bool' },
+  ],
+  [CommandType.V4_SWAP]: [{ name: 'command', type: 'bytes' }],
 
   // Token Actions and Checks
-  [CommandType.WRAP_ETH]: ['address', 'uint256'],
-  [CommandType.UNWRAP_WETH]: ['address', 'uint256'],
-  [CommandType.SWEEP]: ['address', 'address', 'uint256'],
-  [CommandType.TRANSFER]: ['address', 'address', 'uint256'],
-  [CommandType.PAY_PORTION]: ['address', 'address', 'uint256'],
-  [CommandType.BALANCE_CHECK_ERC20]: ['address', 'address', 'uint256'],
+  [CommandType.WRAP_ETH]: [
+    { name: 'recipient', type: 'address' },
+    { name: 'amountMin', type: 'uint256' },
```

Comment:
i know its named amountMin in the contract but thats a mistake 😆 
its actually just `amount`

Reply:
it goes into this function which is just amount
https://github.com/Uniswap/universal-router/blob/f95928955d26ec2b6a74058c63fa8c2747c734f6/contracts/modules/Payments.sol#L111

Reply:
ty fixed!